src/modals/modal-overlay/modal-overlay.component.js

```diff
@@ -0,0 +1,131 @@
+angular.module('patternfly.modals')
+  .component('pfModalOverlay', {
+    bindings: {
+      showModal: '<',
+      close: "&onClose",
+      modalId: '=',
+      modalTitle: "=",
+      titleId: "=?",
+      hideCloseIcon: "<?",
+      backgroundClose: "<?",
+      onBackgroundClick: "=?",
+      isForm: "<?",
+      modalBodyTemplate: "=",
+      modalBodyScope: "=?",
+      actionButtons: "<"
+    },
+    templateUrl: 'modals/modal-overlay/modal-overlay.html',
+    controller: function ( $uibModal ) {
+      'use strict';
+
+      var ctrl = this;
+
+      ctrl.open = function () {
+        $uibModal.open({
+          component: 'pfModalOverlayContent',
+          backdrop: ctrl.backgroundClose ? true : 'static',
+          resolve: {
+            modalId: function () {
+              return ctrl.modalId;
+            },
+            titleId: function () {
+              return ctrl.titleId || "modalTitle";
+            },
+            modalTitle: function () {
+              return ctrl.modalTitle;
+            },
+            hideCloseIcon: function () {
+              return ctrl.hideCloseIcon;
+            },
+            onBackgroundClick: function() {
+              return ctrl.onBackgroundClick;
+            },
+            isForm: function() {
+              return ctrl.isForm;
+            },
+            modalBodyTemplate: function() {
+              return ctrl.modalBodyTemplate;
+            },
+            modalBodyScope: function() {
+              return ctrl.modalBodyScope;
+            },
+            actionButtons: function () {
+              return ctrl.actionButtons;
+            }
+          }
+        })
+          .result.then(
+          function (dismissCause) {
+            ctrl.close({'dismissCause': dismissCause}); // closed
```

Comment:
Is there a reason to use a named parameter here?

Reply:
It how we pass the `dismissCause` back to user defined callback:  `$scope.onClose2 = function(dismissCause) ...`